tools/packaging/build_wheel.py

```diff
@@ -140,13 +140,61 @@ class Builder:
     # The python interpreter that we should be using
     interpreter: str
 
-    def __init__(self, interpreter: str) -> None:
+    def __init__(self, interpreter: str, log_destination: str) -> None:
         self.interpreter = interpreter
+        self.log_destination = log_destination
 
     def setup_py(self, cmd_args: list[str]) -> bool:
-        return (
-            run_cmd([self.interpreter, str(SETUP_PY_PATH), *cmd_args]).returncode == 0
-        )
+        interpreter_version_str = interpreter_version(self.interpreter)
+        captured_output = []
+
+        # Use Popen for real-time streaming
+        with subprocess.Popen(
+            [self.interpreter, str(SETUP_PY_PATH), *cmd_args],
+            stdout=subprocess.PIPE,
+            stderr=subprocess.STDOUT,
+            universal_newlines=True,
+            bufsize=0,  # Unbuffered for real-time output
+        ) as process:
+            if process.stdout is None:
+                raise ValueError("Failed to start process")
+            if self.log_destination:
+                logger.info("Logging to %s", self.log_destination)
+            counter = 0
+            # Stream output line by line
+            for line in process.stdout:
+                if not self.log_destination:
+                    print(
+                        line, end=""
+                    )  # Print to console in real time if we're not logging to a file
+                    sys.stdout.flush()  # Force flush to ensure immediate output
+                else:
+                    # print a dot for every 10 lines to indicate progress
+                    if counter % 10 == 0:
+                        print(".", end="")
+                        sys.stdout.flush()  # Force flush for dots too
```

Comment:
You can do the flushing with each print statement. No need for two lines per print command

```suggestion
                        print(".", end="", flush=True)
```